### PR TITLE
S3 Command port to Pycog

### PR DESCRIPTION
**Pull request by lhaskins**
This PR covers porting the S3 commands over to use pycog.

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -41,7 +64,11 @@
             "when command is mist:ec2-tags must have mist:manage-tags",
             "when command is mist:s3-buckets must have mist:view",
             "when command is mist:s3-buckets with arg[0] == 'list' must have mist:view",
-            "when command is mist:s3-buckets with (arg[0] == 'delete' or arg[0] == 'rm') must have mist:destroy"],
+            "when command is mist:s3-buckets with (arg[0] == 'delete' or arg[0] == 'rm') must have mist:destroy",
```
Comment (kevsmith) [inline config.json]: Why change from `s3-buckets` to `s3-buckets-list`?

**Comment by kevsmith** [comment]
Overall :+1: on changes aside from my question re: rename from `s3-buckets` to `s3-buckets-list`.

**Comment by kevsmith** [comment]
Are these changes compatible with your recently merged [pycog](https://github.com/cog-bundles/pycog) PR?

File: config.json (hunk 1 above)
Comment (lhaskins) [inline config.json]: It's not a change; it's in addition too. `s3-buckets` only deals with buckets. `s3-bucket-list` deals with what's in the buckets. Maybe that's a bad name for the command. How about `s3-bucket_files`?

**Comment by lhaskins** [comment]
Yes, I made a few changes this morning after seeing your comments in pycog

**Pull request merged**